File: web/src/helpers/ReadableUnitHelper.test.js
import readableUnitHelper from './ReadableUnitHelper'

test('should display a byte number', () => {
  const readableSize = readableUnitHelper(10);
  expect(readableSize).toEqual('10 B');
});

test('should display a KiloByte number', () => {
  const readableSize = readableUnitHelper(10240);
  expect(readableSize).toEqual('10 KB');
});

test('should display a MegaByte number', () => {
  const readableSize = readableUnitHelper(10485760);
  expect(readableSize).toEqual('10 MB');
});

test('should display a GigaByte number', () => {
  const readableSize = readableUnitHelper(10737418240);
  expect(readableSize).toEqual('10 GB');
});

File: modules/api/src/test/kotlin/info/mdamour/gottaGoFinder/helpers/ReadableUnitHelperTest.kt
package info.mdamour.gottaGoFinder.helpers

import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

internal class ReadableUnitHelperTest {
    @Test
    fun `when services should return byte`() {
        val displayedMessage = ReadableUnitHelper(10).display()
        assertTrue(displayedMessage == "10 B")
    }

    @Test
    fun `when services should return Kilobyte`() {
        val displayedMessage = ReadableUnitHelper(10240).display()
        assertTrue(displayedMessage == "10.0 KB")
    }

    @Test
    fun `when services should return Megabyte`() {
        val displayedMessage = ReadableUnitHelper(10485760).display()
        assertTrue(displayedMessage == "10.0 MB")
    }

    @Test
    fun `when services should return Gigabyte`() {
        val displayedMessage = ReadableUnitHelper(10737418240).display()
        assertTrue(displayedMessage == "10.0 GB")
    }
}